### ISSUE TITLE
[a11y]: Text input placeholder text , color contrast

### ISSUE THREAD

**Issue by psurana2** [carbon-design-system/carbon#16196]
### Package

@carbon/react

### Browser

Chrome, Firefox

### Operating System

MacOS

### Package version

v1.50.0

### React version

v17.0.2

### Automated testing tool and ruleset

Manual

### Description

The place holder text color for a text input is not readable easily , need to have a clear differentiation between foreground and background color ( g90/100 theme)

### Reproduction/example

https://stackblitz.com/edit/github-ydhxsh?file=src%2FApp.jsx

### Steps to reproduce

1. Run the code 
2. Observe the color difference between the foreground and background 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by laurenmrice** [comment]
Moving forward, we recommend not using placeholder text. It does not need to pass accessibility contrast because it disappears after interacting with the text input. We recommend using the helper text with information to help the user fill out the text input.
